fdc.py

```diff
@@ -0,0 +1,214 @@
+
+class FDCResponseList(list):
+    """Wrapper type for adding the success field to response lists."""
+
+    def __getitem__(self, i):
+        if i == "success":
+            return True
+
+        return list.__getitem__(self, i)
+
+    def __contains__(self, o):
+        if o == "success":
+            return True
+
+        return list.__contains__(self, o)
+
+
+def _response(raw):
+    # returning an error object matching the API error style for request exceptions
+    if isinstance(raw, RequestException):
+        return {
+            "success": False,
+            "error": {"code": "REQUEST_EXCEPTION", "message": str(raw)},
+        }
+
+    # ensuring the "success" field is always present
+    if isinstance(raw, list):
+        return FDCResponseList(raw)
+
+    raw["success"] = "error" not in raw
+    return raw
+
+
+def _get(endpoint, params=None):
+    if _SECRETS["FDC_KEY"] is None:
+        raise Exception("FDC key has not been set")
+
+    url = f"{_FDC_URL}/{endpoint}"
+
+    params = params or {}
+    params["api_key"] = _SECRETS["FDC_KEY"]
+
+    try:
+        response = requests.get(url, params=params)
+        response.raise_for_status()
+        return _response(response.json())
+    except RequestException as exc:
+        return _response(exc)
+
+
+def _post(endpoint, data=None):
+    if _SECRETS["FDC_KEY"] is None:
+        raise Exception("FDC key has not been set")
+
+    url = f"{_FDC_URL}/{endpoint}"
+    params = {"api_key": _SECRETS["FDC_KEY"]}
+
+    try:
+        response = requests.post(
+            url,
+            params=params,
+            data=json.dumps(data or {}),
+            headers={"Content-Type": "application/json"},
+        )
+        response.raise_for_status()
+        return _response(response.json())
+    except RequestException as exc:
+        return _response(exc)
+
+
+def _assign_search_info(data, page, per_page, sort, sort_direction):
+    # if sort is a key in _SORT_VALUES, get the value.
+    # otherwise, use the supplied sort parameter as the value.
+    sort = _SORT_VALUES.get(sort, sort)
+
+    if page != 1:
+        data["pageNumber"] = page
+
+    if per_page != 50:
+        data["pageSize"] = per_page
+
+    if sort is not None:
+        data["sortBy"] = sort
+
+    if sort_direction is not None:
+        data["sortOrder"] = sort_direction
+
+    data["dataType"] = _DATA_TYPES
+
+
+# ultimately, the functions below will probably return an
+# object instead of dict so it's easier to work with the results.
+
+
+def get_food_by_id(fdc_id, nutrient_numbers=None, abridged=False):
+    """
+    Retrieves information about a food given its FDC ID.
+
+    Args:
+        - fdc_id: The FDC ID of the food to retrieve.
+        - nutrient_numbers: An optional list of up to 25 nutrients to retrieve.
+            If the food has no matches with the supplied nutrients,
+            the "foodNutrients" field in the response will be empty.
+        - abridged: Whether to return abridged results.
```

Comment:
What is meant by abridged results? Like in what format will these abridged results output?

Reply:
See the [API documentation](https://fdc.nal.usda.gov/api-spec/fdc_api.html). The relevant section is the "Models" section at the bottom of the page. Essentially, it's a version of the results which leaves out a lot of fields. We may end up not needing it for our project, but it seems like it might be useful if we implement favorite foods. We don't have to retrieve *all* of the information about saved foods, so it would help speed up the request since there's less data to receive.